parser/src/parser.rs

```diff
@@ -202,14 +204,22 @@ impl<'a> Parser<'a> {
 
         let pivot = self.cursor.peek().token_type;
         match pivot {
-            RoundedLeftBracket => self.parenthesis().map(Expr::Parenthesis),
+            RoundedLeftBracket => self.lambda_or_parentheses(),
             CurlyLeftBracket => self.block().map(Expr::Block),
             Not => self.not(Parser::next_value),
 
             //expression that can also be used as values
             If => self.parse_if(Parser::value).map(Expr::If),
             Match => self.parse_match(Parser::value).map(Expr::Match),
             Identifier if self.may_be_at_programmatic_call_start() => self.programmatic_call(),
+            Identifier
+                if self
+                    .cursor
+                    .lookahead(any().then(blanks().then(of_type(FatArrow))))
+                    .is_some() =>
+            {
+                self.parse_lambda_definition().map(Expr::LambdaDef)
+            }
```

Comment:
These lines are not covered by any test.
```rs
#[test]
fn lambda_one_arg() {
    let source = Source::unknown("calc(n => $n * $n)");
    let parsed = parse(source).expect("Failed to parse.");
    assert_eq!(
        parsed,
        vec![Expr::ProgrammaticCall(ProgrammaticCall {
            name: "calc",
            arguments: vec![Expr::LambdaDef(LambdaDef {
                args: vec![TypedVariable {
                    name: "n",
                    ty: None,
                }],
                body: Box::new(Expr::Binary(BinaryOperation {
                    left: Box::new(Expr::VarReference(VarReference { name: "n" })),
                    op: BinaryOperator::Times,
                    right: Box::new(Expr::VarReference(VarReference { name: "n" })),
                })),
            })],
            type_parameters: vec![],
        })]
    );
}
```

Reply:
done